### DISCUSSION TITLE
question about exports

### DISCUSSION THREAD

**Discussion by vajonam** [blakeblackshear/frigate#7894] [General]
When I select a `09/19/2023 12:00 AM` to `09/19/2023 11:59 PM" I get an error with no videos for that time line in the logs

```
2023-09-20 11:57:01.891488254  [2023-09-20 11:57:01] frigate.record.export          ERROR   : Failed to export recording for command ffmpeg -hide_banner -y -protocol_whitelist pipe,file,http,tcp -f concat -safe 0 -i /dev/stdin -c:v libx264 -preset:v ultrafast -tune:v zerolatency -vf setpts=0.04*PTS -r 30 -an /media/frigate/exports/in_progress.front@2023_09_19_00:00__2023_09_19_23:59.mp4
2023-09-20 11:57:01.891646659  [2023-09-20 11:57:01] frigate.record.export          ERROR   : [http @ 0x558efc4bb380] HTTP error 502 Bad Gateway
2023-09-20 11:57:01.891653295  [concat @ 0x558efc4a7b00] Impossible to open 'http://127.0.0.1:5000/vod/front/start/1695096000/end/1695103200/index.m3u8'
```

but if change that to `09/18/2023 12:00 AM`  to `09/18/2023 11:59 PM` the export works. The day before, why is this? what am I missing? 

2nd question is can I enable hw accel for the export, I have GPUs that can do the work. I am already doing this in an external script will be nice to have it all within frigate.

**Comment by NickM-27** [comment]
it can depend on your record retention mode, I use `all` for 24/7 recording so that is what was mainly tested on. I have a PR up that should fix things when there are gaps between recording segments https://github.com/blakeblackshear/frigate/pull/7846

> 2nd question is can I enable hw accel for the export, I have GPUs that can do the work. I am already doing this in an external script will be nice to have it all within frigate.

hw accel only applies to timelapse, there is nothing for the hardware to accelerate for the normal exports. For the timelapses you just need to have an ffmpeg preset set globally for `ffmpeg -> hwaccel_args`

**Reply by vajonam** [reply]
thanks.

**Comment by vajonam** [comment]
I have use case that executes a shell script creates exports on a daily basis and save is in a folder that is viewable on  web page. Is the long range idea to have something like this, where can save a given number of days of timelapses and they are created early every day?  right now all this is done via crontab and shell scripts and have standalone html that reads a folder and generates a list of files. 

https://github.com/blakeblackshear/frigate/assets/152501/7acba29f-904d-4f7a-b1af-9519a0427b61


if so I can maybe contribute the exports viewer that is more featured than the initial implementation.

**Reply by NickM-27** [reply]
> Is the long range idea to have something like this, where can save a given number of days of timelapses and they are created early every day? right now all this is done via crontab and shell scripts and have standalone html that reads a folder and generates a list of files.

I think the scheduling is something that would be better left outside of frigate. It would be trivial to add a service in HA which creates an export and can be automated there and of course a crontab automation that calls frigate's export api would be pretty simple as well.

> if so I can maybe contribute the exports viewer that is more featured than the initial implementation.

not sure what you have in mind, but I already have a PR up that allows playback and deleting of exports from the webUI

**Reply by vajonam** [reply]
cool will go check that out! nice!

**Comment by weitheng** [comment]
@vajonam Can you please share your shell script that creates the exports daily? I want to do something similar and I am not share how to do it

**Reply by vajonam** [reply]
https://gist.github.com/vajonam/b2697ce137781116707392f95891ddf1

**Reply by vajonam** [reply]
its reasonably complex, it uses ffmpeg in container with hardware acceleration. if you pass it a date it will use that, if not it will generate the export for yesterday. if you dont have hardwarware accel you will have to disable hw in ffmpeg command line.

**Reply by bozoboo** [reply]
Thanks for this script.

Are you still running it?
